File: 2022/06/17/Stella/Q9_Palindrome_Number.java
class Solution{
    public boolean isPalindrome(int x) {
        
        if(x<0)return false;
        for(int i=0; i<String.valueOf(x).length();i++){
            if(String.valueOf(x).charAt(i)!=String.valueOf(x).charAt(String.valueOf(x).length()-i-1)){
                 return false;
            }

        }
        return true;
    }
}

File: 2022/06/17/Guan/Q9_Palindrome_Number.cpp
#include <iostream>
#include <string>

using namespace std;

class Solution {
public:
    bool isPalindrome(int x) {
        string theString = to_string(x);
        if(x < 0){
            return false;
        }
        // else if(theString.length()%2 == 0){
        //     //if the string have a size even..


        // }
        else{
            string readForward = "";
            string readBackward = "";
            int size_ = theString.length();
            for(int i = 0; i < size_; i++){
                //store them both front and back;
                readForward += theString[i];
                readBackward += theString[size_-i-1];
                //if they are not the same, abort:
                if(readForward != readBackward){
                    // cout << "Forward: " << readForward;
                    // cout << "Backward: " << readBackward;
                    return false;
                }
            }
            return true;
        }
    }
};